src/coreclr/nativeaot/Runtime/loongarch64/PInvoke.S

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+#include <unixasmmacros.inc>
+#include "AsmOffsets.inc"
+
+.global RhpTrapThreads
+
+// Note: these must match the defs in PInvokeTransitionFrameFlags defined in rhbinder.h
+PTFF_SAVE_SP            = 0x00000200
+
+// Bit position for the flags above, to be used with andi+beq/bne instructions
+PTFF_THREAD_ABORT_BIT   = 36
+
+//
+// RhpPInvoke
+//
+// IN:  a0: address of pinvoke frame
+//
+// This helper assumes that its callsite is as good to start the stackwalk as the actual PInvoke callsite.
+// The codegenerator must treat the callsite of this helper as GC triggering and generate the GC info for it.
+// Also, the codegenerator must ensure that there are no live GC references in callee saved registers.
+//
+
+NESTED_ENTRY RhpPInvoke, _TEXT, NoHandler
+        st.d  $fp, $a0, OFFSETOF__PInvokeTransitionFrame__m_FramePointer
+        st.d  $ra, $a0, OFFSETOF__PInvokeTransitionFrame__m_RIP
+        ori  $t0, $sp, 0
+        st.d  $t0, $a0, OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs
```

Comment:
Can this `st.d` operate on `$sp` directly?

Reply:
Yes, I think you are right.
Thanks!